src/Illuminate/Support/Arr.php

```diff
@@ -331,7 +331,19 @@ public static function isAssoc(array $array)
      */
     public static function only($array, $keys)
     {
-        return array_intersect_key($array, array_flip((array) $keys));
+        $attributeData = [];
+
+        foreach ((array) $keys as $key) {
+            $keyValue = data_get($array, $key, '__missing__');
+
+            if ($keyValue == '__missing__') {
+                continue;
+            }
+
+            data_set($attributeData, $key, $keyValue);
```

Comment:
Do your $key contain "*" ? I think the `data_set` is too "useful", `Arr::set` should work.